modules/sw360/sw360-workflow/src/main/java/org/eclipse/sw360/antenna/sw360/SW360MetaDataUpdater.java

```diff
@@ -108,15 +112,74 @@ public boolean isUploadSources() {
     public SW360Release uploadAttachments(SW360Release sw360Release, Map<Path, SW360AttachmentType> attachments) {
         AttachmentUploadRequest.Builder<SW360Release> builder = AttachmentUploadRequest.builder(sw360Release);
         for (Map.Entry<Path, SW360AttachmentType> e : attachments.entrySet()) {
-            builder = builder.addAttachment(e.getKey(), e.getValue());
+            if (needUpload(sw360Release, e.getKey(), e.getValue())) {
+                builder = builder.addAttachment(e.getKey(), e.getValue());
+            }
         }
 
-        AttachmentUploadResult<SW360Release> result = releaseClientAdapter.uploadAttachments(builder.build());
-        LOGGER.debug("Result of attachment upload operation: {}", result);
-        if (!result.isSuccess()) {
-            LOGGER.error("Failed to upload attachments: {}", result.failedUploads());
+        AttachmentUploadRequest<SW360Release> uploadRequest = builder.build();
+        if (!uploadRequest.getItems().isEmpty()) {
+            AttachmentUploadResult<SW360Release> result = releaseClientAdapter.uploadAttachments(uploadRequest);
+            LOGGER.debug("Result of attachment upload operation: {}", result);
+            if (!result.isSuccess()) {
+                LOGGER.error("Failed to upload attachments: {}", result.failedUploads());
+            }
+            return result.getTarget();
+        }
+        return sw360Release;
+    }
+
+    /**
+     * Calculates the SHA-1 hash for the given local attachment file. This is
+     * used to determine whether a modified attachment file needs to be
+     * uploaded.
+     *
+     * @param path the path to the local file
+     * @return the SHA-1 hash of this file
+     * @throws SW360ClientException if an error occurs
+     */
+    String calculateAttachmentHash(Path path) {
+        return SW360AttachmentUtils.calculateSha1Hash(path);
```

Comment:
isn't this method a bit superfluous?

Reply:
It is used by tests to check whether exceptions are properly handled.